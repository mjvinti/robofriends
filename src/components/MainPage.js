import React, { Component } from 'react';

import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundary from '../components/ErrorBoundary';
import Header from '../components/Header';

import './MainPage.css';

class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots();
    }

    render() {
        const { isPending, onSearchChange, robots, searchField } = this.props;
        const filteredRobots = robots.filter((robot) =>
            robot.name.toLowerCase().includes(searchField.toLowerCase())
        );

        return isPending
            ? <h1>Loading</h1>
            : (
                <div className='tc'>
                    <Header />
                    <SearchBox searchChange={onSearchChange} />
                    <Scroll>
                        <ErrorBoundary>
                            <CardList robots={filteredRobots} />
                        </ErrorBoundary>
                    </Scroll>
                </div>
            );
    }
}

export default MainPage;