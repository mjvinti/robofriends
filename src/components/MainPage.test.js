import React from 'react';
import { shallow } from 'enzyme';

import MainPage from './MainPage';

let mockProps, wrapper;
beforeEach(() => {
    mockProps = {
        onRequestRobots: jest.fn(),
        onSearchChange: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    };
    wrapper = shallow(<MainPage {...mockProps} />)
});

it('renders MainPage', () => {
    expect(wrapper).toMatchSnapshot();
});

it('renders MainPage when `isPending` is `true`', () => {
    expect(wrapper.setProps({ ...mockProps, isPending: true })).toMatchSnapshot();
});

it('renders filtered robots', () => {
    const robots = [{ name: 'Steve' }, { name: 'Stu' }, { name: 'Eric' }];
    const searchField = 'st';
    expect(wrapper.setProps({ ...mockProps, robots, searchField })).toMatchSnapshot();
});