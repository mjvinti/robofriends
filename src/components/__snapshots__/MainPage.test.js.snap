// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox
    searchChange={[MockFunction]}
  />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;

exports[`renders MainPage when \`isPending\` is \`true\` 1`] = `
<h1>
  Loading
</h1>
`;

exports[`renders filtered robots 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox
    searchChange={[MockFunction]}
  />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={
          Array [
            Object {
              "name": "Steve",
            },
            Object {
              "name": "Stu",
            },
          ]
        }
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
